server/routes/editDraft.ts

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-restricted-syntax */
+import { Request, Response, NextFunction } from 'express';
+import lookupAddressIsOwnedByUser from '../util/lookupAddressIsOwnedByUser';
+import { factory, formatFilename } from '../../shared/logging';
+import lookupCommunityIsVisibleToUser from '../util/lookupCommunityIsVisibleToUser';
+
+const log = factory.getLogger(formatFilename(__filename));
+
+export const Errors = {
+  NoId: 'Must supply draft ID.',
+  NotOwner: 'User does not have permission to edit this thread.',
+  NotFound: 'Draft not found.'
+};
+
+const editDraft = async (models, req: Request, res: Response, next: NextFunction) => {
+  const [chain, community] = await lookupCommunityIsVisibleToUser(models, req.body, req.user, next);
+  const author = await lookupAddressIsOwnedByUser(models, req, next);
+  if (!req.body.id) {
+    return next(new Error(Errors.NoId));
+  }
+
+  const { id, title, body, tag } = req.body;
+
+  const attachFiles = async () => {
+    if (req.body['attachments[]'] && typeof req.body['attachments[]'] === 'string') {
+      await models.OffchainAttachment.create({
+        attachable: 'draft',
+        attachment_id: id,
+        url: req.body['attachments[]'],
+        description: 'image',
+      });
+    } else if (req.body['attachments[]']) {
+      await Promise.all(req.body['attachments[]'].map((url) => models.OffchainAttachment.create({
+        attachable: 'draft',
+        attachment_id: id,
+        url,
+        description: 'image',
+      })));
+    }
+  };
+
+  try {
+    const userOwnedAddresses = await req.user.getAddresses();
+    const draft = await models.DiscussionDraft.findOne({
+      where: {
+        id,
+      },
+      include: [
+        models.Address,
+        models.OffchainAttachment
+      ]
+    });
+    if (!draft) return next(new Error(Errors.NotFound));
+    if (userOwnedAddresses.filter((addr) => addr.verified).map((addr) => addr.id).indexOf(draft.author_id) === -1) {
```

Comment:
can we pass userOwnedAddressIDs to the DiscussionDraft.findOne query instead of doing this check here?

Reply:
Sure! This is our default pattern right now, but I can update all of the routes if you'd like? 
![image](https://user-images.githubusercontent.com/22281010/85328979-85b3ec80-b486-11ea-828a-dfe08ed7602c.png)

Reply:
It looks like we only use the other pattern in `server/routes/deleteThread.ts` but I think it's better both in conciseness and performance

Reply:
taken care of